### PR TITLE
storagecluster: Update uninstall error as an events

### PR DESCRIPTION
**Pull request by iamniting**
Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

**Comment by iamniting** [comment]
@raghavendra-talur This is the shortest way with minimum lines of change I can think of to update the last status of uninstall events. Let me know your thoughts on this.

**Comment by iamniting** [comment]
Depends on https://github.com/openshift/ocs-operator/pull/1060. Pls do not merge this one before #1060

File: controllers/util/events.go [hunk 1]
```diff
@@ -6,4 +6,7 @@ const (
 
 	// EventReasonValidationFailed is used when the StorageCluster spec validation fails
 	EventReasonValidationFailed = "FailedValidation"
+
+	// EventReasonUninstallFailed is used when the StorageCluster uninstall fails
+	EventReasonUninstallFailed = "FailedUninstall"
```
Comment (raghavendra-talur) [inline controllers/util/events.go:11]: I think the event is generated in two cases:
1. valid error where we know for sure that it won't progress
2. error where we are just waiting for the uninstall to progress in a sub component.
Unfortunately, you don't get to know the category in the caller.

Therefore, "FailedUninstall" might not be the best wording to use here. How about 
* UninstallInProgress
* UninstallPending

Comment (iamniting) [reply controllers/util/events.go]: I like UninstallPending, Updated the same

**Review by raghavendra-talur — APPROVED** [review]

**Review by jarrpa — APPROVED** [review]
/lgtm

**Pull request merged**